configuration/klippy/beacon_adaptive_heat_soak.py

```
@@ -0,0 +1,722 @@
+			# Wait up to 5 seconds for 1000 good samples to be collected
+			# This is a bit arbitrary, but it should be enough to ensure the beacon is ready.
+			start_time = eventtime = self.reactor.monotonic()
+			while good_samples < 1000 and (eventtime - start_time) < 5:
+				if self.printer.is_shutdown():
+					raise self.printer.command_error(f"{self.name}: Printer is shutting down")
+				eventtime = self.reactor.pause(eventtime + 0.1)
+
+		logging.info(f"{self.name}: Prepared for sampling, collected {good_samples} good samples and {bad_samples} bad samples (total {good_samples+bad_samples} samples).")
+
+		if good_samples < 1000:
+			raise self.printer.command_error(f"Failed to prepare beacon for sampling, timed out waiting for good samples. Beacon must be calibrated and positioned correctly before running this command.")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)